signer/src/transaction_signer.rs

```diff
@@ -0,0 +1,69 @@
+//! # Transaction signer
+//!
+//! This module contains the transaction signer, which is
+//! responsible for preparing and participating in signing rounds.
+//!
+//! For more details, see the [`EventLoop`] documentation.
+
+#[cfg_attr(doc, aquamarine::aquamarine)]
+/// # Transaction signer event loop
+///
+/// This struct contains the implemetnation of the transaction signer logic.
+/// The event loop subscribes to storage update notifications from the block observer,
+/// and listens to signer messages from the signer entwork.
+///
+/// ## On storage update
+/// When the signer receives a storage update notification, it must go over each of the pending
+/// requests and decide wheter to accept or reject it. The decision is then persisted
+/// and broadcast to the other signers. The following flowchart illustrates the flow.
+///
+/// ```mermaid
+/// flowchart TD
+///     SU{Storage update notification} --> FPR(Fetch pending requests)
+///     FPR --> NR(Next request)
+///     NR --> |deposit/withdraw| DAR(Decide to accept/reject)
+///     NR ----> |none| DONE{Done}
+///     DAR --> PD(Persist decision)
+///     PD --> BD(Broadcast decision)
+///     BD --> NR
+/// ```
+///
+/// ## On signer message
+/// When the signer receives a signer message, it needs to do a few different things
+/// depending on the type of the message.
+///
+/// - **Signer decision**: When receiving a signer decision, the transaction signer
+/// only need to persist the decision to the signer storage.
+/// - **Stacks sign request**: When receiving a request to sign a stacks transaction,
+/// the signer must verify that it has decided to sign the transaction, and if positive,
+/// send a transaction signature back over the network.
+/// - **Bitcoin sign request**: When receiving a request to sign a bitcoin transaction,
+/// the signer must verify that it has decided to accept all request that the
+/// transaction fulfills. Once verified, the transaction signer creates a dedicated
+/// WSTS state machine to participate in a signing round for this transaction. Thereafter,
+/// the signer sends a bitcoin transaction sign ack message back over the network to signal
+/// its readiness.
+/// - **WSTS message**: When receiving a WSTS message, the signer will look up the corresponding
+/// state machine and relay the WSTS message to it.
```

Comment:
```suggestion
/// - **WSTS message**: When receiving a WSTS message, the signer will look up the corresponding
/// internal state machine process and relay the WSTS message to it.
```

Reply:
I don't think it's a separate process, so perhaps "relay" is the wrong word here.

Reply:
`dispatch`?

Reply:
Sounds good, thanks for suggesting :)